lua/lean/_util.lua

```diff
@@ -5,6 +5,15 @@ local a = require("plenary.async")
 
 local M = {}
 
+---@generic T
+---@param dest T[]
+---@param src T[]
+function M.insert_all(dest, src)
```

Comment:
Isn't this vim.list_extend?

Reply:
Yes, you're right.  I usually try to avoid the `vim.*` functions because they have lots of performance issues (see e.g. https://github.com/neovim/neovim/pull/16197 where list_extend was removed for that reason).  But this is not a performance critical part, so let's use list_extend.